base/char.jl

```diff
@@ -24,6 +61,11 @@ function isoverlong(c::Char)
     is_overlong_enc(u)
 end
 
+# fallback: other AbstractChar types, by default, are assumed
+#           not to support malformed or overlong encodings.
+ismalformed(c::AbstractChar) = false
```

Comment:
Is it really a good idea to define these fallbacks? They can be wrong for some types, and they are easy to implement. If you implement your own `AbstractChar` type, it's good to think about this question precisely.

Reply:
All of the proposed `AbstractChar` types I've ever seen, other than `Char`, don't support malformed characters.   So it makes sense to me to make this the default.  You can always override it.

Another reason to make this the default is that `ismalformed(c)` returns `true`, you need to define additional methods to decode the character.